packages/cfd/src/Components/__tests__/cfd-demo-account-display.spec.js

```diff
@@ -0,0 +1,361 @@
+            onSelectAccount: jest.fn(),
+            openAccountNeededModal: jest.fn(),
+            openAccountTransfer: jest.fn(),
+            openPasswordManager: jest.fn(),
+            platform: 'mt5',
+            residence: 'id',
+            standpoint: {
+                financial_company: 'svg',
+                gaming_company: 'svg',
+                iom: false,
+                malta: false,
+                maltainvest: false,
+                svg: true,
+            },
+        };
+    });
+
+    const mt5_demo_financial_account = {
+        account_type: 'demo',
+        balance: 10000,
+        country: 'id',
+        currency: 'USD',
+        display_balance: '10000.00',
+        display_login: '20103240',
+        email: 'name@domain.com',
+        group: 'demo\\p01_ts02\\financial\\svg_std_usd',
+        landing_company_short: 'svg',
+        leverage: 1000,
+        login: 'MTD20103240',
+        market_type: 'financial',
+        name: 'Name LastName',
+        server: 'p01_ts02',
+        server_info: {
+            environment: 'Deriv-Demo',
+            geolocation: {
+                group: 'all',
+                location: 'N. Virginia',
+                region: 'US East',
+                sequence: 1,
+            },
+            id: 'p01_ts02',
+        },
+        sub_account_type: 'financial',
+    };
+
+    const mt5_demo_financial_stp = {
+        account_type: 'demo',
+        balance: 10000,
+        country: 'id',
+        currency: 'USD',
+        display_balance: '10000.00',
+        display_login: '20103241',
+        email: 'name@domain.com',
+        group: 'demo\\p01_ts02\\financial\\labuan_stp_usd',
+        landing_company_short: 'labuan',
+        leverage: 100,
+        login: 'MTD20103241',
+        market_type: 'financial',
+        name: 'Name LastName',
+        server: 'p01_ts02',
+        server_info: {
+            environment: 'Deriv-Demo',
+            geolocation: {
+                group: 'all',
+                location: 'N. Virginia',
+                region: 'US East',
+                sequence: 1,
+            },
+            id: 'p01_ts02',
+        },
+        sub_account_type: 'financial_stp',
+    };
+
+    const dxtrade_demo_synthetic_account = {
+        account_id: 'DXD1096',
+        account_type: 'demo',
+        balance: 10000,
+        currency: 'USD',
+        display_balance: '10000.00',
+        display_login: 'DXD1096',
+        enabled: 1,
+        landing_company_short: 'svg',
+        login: '374',
+        market_type: 'synthetic',
+        platform: 'dxtrade',
+    };
+
+    const checkAccountCardsRendering = tested_case => {
+        const component_testid = 'dt_cfd_demo_accounts_display';
+        const first_account_card = 'Synthetic';
+        const second_account_card = {
+            eu: 'CFDs',
+            non_eu: 'Financial',
+        };
+        const third_account_card = 'Financial STP';
+
+        if (tested_case === TESTED_CASES.LOADING) {
```

Comment:
the reason why I chose `if else` instead of `switch` is that, as you can see, we have to do repeated actions/check the same things for 2/3 different cases, so we'll have a lot of duplicated lines. Due to dividing it into cases, I'm avoiding duplication here. 🙂

Reply:
yeah, it prevents duplication but add complexity. so it's on you which one to choose :D

Reply:
I see, thank you so much for the suggestion 😊